src/zjs_ipm.c

```diff
@@ -98,4 +98,15 @@ void zjs_ipm_register_callback(uint32_t msg_id, ipm_callback_t cb)
 #endif
 }
 
+void zjs_ipm_free_callbacks() {
+    #ifdef CONFIG_X86
```

Comment:
so CONFIG_X86 means it's only for the x86 side and not ARC, so I think it's correct here, the whole IPM is conditionally inside CONFIG_BOARD_ARDUINO_101, but CONFIG_X86 and CONFIG_ARC further separate between them, since ARC doesn't have any mem alloc capababilities, it makes sense to fall into X86 only.